src/main/java/com/zipcodewilmington/looplabs/IntegerDuplicateDeleter.java

```diff
@@ -1,8 +1,105 @@
 package com.zipcodewilmington.looplabs;
 
+import java.util.Arrays;
+
 /**
  * Created by leon on 1/29/18.
  * @ATTENTION_TO_STUDENTS You are forbidden from modifying the signature of this class.
  */
 public final class IntegerDuplicateDeleter extends DuplicateDeleter<Integer> {
-}
+    //constructor
+    public IntegerDuplicateDeleter(Integer[] intArray) {
+        super(intArray);
+    }
+
+    Integer[] anIntArray = Arrays.copyOf(this.array, this.array.length);
+
+    @Override
+    public Integer[] removeDuplicates(int maxNumberOfDuplications) {
+
+        int noDuplicates = 0;
```

Comment:
this variable name is weird. If you're just using it to check if there is or isn't a duplicate, this should be a boolean. If you are checking for the number of duplicates, then it should be called numberOfDuplicates

Reply:
will definitely update this, thank you for the feedback. I've already put this file on my computer again. I want to go through all of the labs again, maybe time myself...highlight the areas I don't know.